api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +565,134 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      if (crs != null) {
+        Preconditions.checkArgument(!crs.isEmpty(), "Invalid CRS: (empty string)");
+        Preconditions.checkArgument(
+            crs.trim().equals(crs), "CRS must not have leading or trailing spaces: '%s'", crs);
+      }
+      this.crs = crs;
+    }
+
+    public static GeometryType crs84() {
+      return new GeometryType(null);
```

Comment:
style: what do you think to have a private GeometryType() constructor that does nothing?  to avoid having null passed in explicitly

Reply:
Added a new zero-arg constructor. Similar change was made for GeographyType.